doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -344,85 +343,96 @@ Rewriting with definitional equality
 .. _performingcomputations:
 
 Performing computations
----------------------------
+-----------------------
+
+.. todo PR: Is the above good title?  Perhaps "Applying reductions" would be better?
```

Comment:
Or even "Applying reduction rules"?

Reply:
Is this correct: conversion rules consist solely of reductions and expansions.  For each reduction there is an associated expansion that is the inverse of the reduction.  If correct, I will add with better wording.

Reply:
That's not quite correct.  My understanding is as follows:  I would say that a "reduction" is generally an evaluation step / normalization step.  All reductions have a corresponding "inverse" expansion, but the expansion is generally not a function; because reductions are many-to-one, expansions are one-to-many, i.e., relations rather than functions.  The corresponding expansion is in fact just the preimage-relation of the reduction.  There are some expansions which are in fact functions, and the corresponding "reduction" is generally called a "contraction" (though it can also be called "reduction", but never "evaluation").  The only example of this I'm aware of is that both functions and records (and, to some extent, inductive types more generally) which have a notion of "eta-expansion" whose inverse is generally called "eta-contraction".  Whether or not the contraction is well-typed depends on some subtleties about subtyping and cumulativity; see https://github.com/HoTT/book/issues/995 for a fascinating discussion.

Reply:
I think I got the terminology untangled.  (There are reductions and expansions (one is the reverse direction of the other, contraction means an invertible reduction.)

Does Coq use eta-reduction/contraction anywhere?  If not, I think I would leave out the functions+records case you mention.